apps/system/js/app_chrome.js

```diff
@@ -709,6 +710,16 @@
 
       this.updateAddToHomeButton();
 
+      // We only change the icon back to the default one if the new page is not
+      // on the same domain than the previous one.
+      // In both cases, we look for the best icon after `mozbrowserloadend`.
+      var origin = this.getOriginFromURL(evt.detail);
```

Comment:
Yes, AFAIK, `new URL()` performs better than the `<a>` tag. That's the reason why I create and reuse the same node object, `this.dummyATag`, over and over again in `getOriginFromURL()`, to avoid possible delays due to object instantiation and garbage collection.
For robustness, I'd definitely recommend use the `<a>` tag method. We don't want to risk throwing exception in this part of the code.